x-pack/plugins/apm/common/service_groups.test.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  isSupportedField,
+  validateServiceGroupKuery,
+  SERVICE_GROUP_SUPPORTED_FIELDS,
+} from './service_groups';
+import {
+  TRANSACTION_TYPE,
+  TRANSACTION_DURATION,
+  SERVICE_FRAMEWORK_VERSION,
+} from './elasticsearch_fieldnames';
+
+const unsupportedFields = [
+  TRANSACTION_TYPE,
+  TRANSACTION_DURATION,
+  SERVICE_FRAMEWORK_VERSION,
+];
+const mockFields = [...SERVICE_GROUP_SUPPORTED_FIELDS, ...unsupportedFields];
+
+describe('service_groups common utils', () => {
+  describe('isSupportedField', () => {
+    it('should only filter supported supported fields for service groups', () => {
+      const supportedFields = mockFields.filter(isSupportedField);
+      expect(supportedFields).toEqual(SERVICE_GROUP_SUPPORTED_FIELDS);
+    });
```

Comment:
nit: Since you are not using `mockFields` or `unsupportedFields` I think they ought to be defined inside the test. I also think it's a little clearer to break them up and test each field separately like:

```suggestion
    it('should allow support fields', () => {
      SERVICE_GROUP_SUPPORTED_FIELDS.map((field) => {
        expect(isSupportedField(field)).toBe(true);
      });
    });

    it('should reject unsupport fields', () => {
      [TRANSACTION_TYPE, TRANSACTION_DURATION, SERVICE_FRAMEWORK_VERSION].map(
        (field) => {
          expect(isSupportedField(field)).toBe(false);
        }
      );
    });
      
```